.github/workflows/ci.yml

```diff
@@ -42,12 +43,16 @@ jobs:
       timeout-minutes: 5
       run: |
         set -x
-        RESULT=" "
-        while [[ "$RESULT" != "  HTTP/1.1 200 OK" ]]; do
-          sleep 1
-          RESULT=$(kubectl exec deployments/frontend -- sh -c "wget --spider -S "http://frontend" 2>&1 | grep 'HTTP/'")
-          echo "front end response: $RESULT"
+        # start fresh loadgenerator pod
+        kubectl delete pod -l app=loadgenerator
+        # wait for requests to come in
+        REQUEST_COUNT="0"
+        while [[ "$REQUEST_COUNT"  -lt "50"  ]]; do
+            sleep 5
+            REQUEST_COUNT=$(kubectl logs -l app=loadgenerator | grep Aggregated | awk '{print $2}')
         done
-        if [[ "$RESULT" != "  HTTP/1.1 200 OK" ]]; then 
+        # ensure there are no errors hitting endpoints
+        ERROR_COUNT=$(kubectl logs -l app=loadgenerator | grep Aggregated | awk '{print $3}' | sed "s/[(][^)]*[)]//g")
+        if [[ "$ERROR_COUNT" -gt "0" ]]; then
```

Comment:
I feel like startup errors or small flakes can instantly fail the smoke test. I wish there was a better way though.

Reply:
Yeah, hopefully over time we can add more robust tests. For now it's easy enough to re-run though (start up errors are reduced by re-starting just the loadgenerator at the start of the test. It's usually pretty stable)